### ISSUE TITLE
Generate tests for codegen

### ISSUE THREAD

**Issue by lmazuel** [Azure/autorest.csharp#1915]
As part of the generation of DP codegen, we would like to take advantage of receiving x-ms-examples of m4 to generate tests files.

The idea is:
- Generate one test per method with x-ms-examples (let's skip operations without them yet)
- Generate models as part of the test folder
- Generate test code:
  - Creates models from the input of the x-ms-examples
  - Call the method with that input
  - Generate a bunch of assertion based on the x-ms-examples output

Important note:
- It is NOT a goal to have those tests running "as-is". Running tests might requires a lot of ARM resource creation, and pre-requisites which are out of the scope of this issue. We HELP service teams by providing 80% of the tests auto-written, and they need to fill the gap

Stretch goal (for latter or follow-up issues):
- Update scenario. How do I regenerate tests if some of them have been manually updated?
- Generate test for operation without x-ms-examples. We could do something like:
   ``` 
   // Create your JSON input for {model name from body type in Swagger}
   my_input = FIXME JSON as String from model serialization
   var result = client.do_the_call(my_input.serialize())
   // Check the result looks like {model name from return type in Swagger}
   ```
   This would help getting started

**Comment by annelo-msft** [comment]
> Generate test for operation without x-ms-examples.

I'm curious why a swagger author would want to opt-out from adding x-ms-examples.  Would that not be the preferred approach?

**Comment by lmazuel** [comment]
> > Generate test for operation without x-ms-examples.
> 
> I'm curious why a swagger author would want to opt-out from adding x-ms-examples. Would that not be the preferred approach?

Yeah, they should not opt-out, I'm just not sure it's enforced enough today

**Comment by lmazuel** [comment]
Example from @mikekistler :
```python
    # Test for DetectChangePoint
    @AnomalyDetectorPreparer()
    def test_detect_change_point(self, anomalydetector_endpoint, anomalydetector_apikey):
        client = self.create_test_client(anomalydetector_endpoint, anomalydetector_apikey)
        with open("tests/examples/ChangePointDetect.json") as f:
            example = json.load(f)
        ret = client.detect_change_point(body=example["parameters"]["body"])
        # structure
        assert ret.keys() == example["responses"]["200"]["body"].keys()
        # assert ret['isChangePoint'] == example["responses"]["200"]["body"]['isChangePoint']
```

TLDR:
- Save the x-ms-examples as-is in a folder loadable from the tests
- Write a test that loads the file and check values

**Comment by ArcturusZhang** [comment]
This feature should have been implemented by https://github.com/Azure/autorest.csharp/pull/3785
Closing
